### PR TITLE
[Maistra-2.6] OSSM-6276 - Tweaks for build and test with envoy-openssl

### PR DESCRIPTION
**Pull request by dcillera**
OSSM-6276 - Maistra Proxy build integration.
Some limitations are still present:
- Only for x86 and aarch64.
- Some failing tests have been commented ("go test" in "pre-submit.sh")

### CODE REVIEW

File: maistra/scripts/update-deps.sh [hunk 1]
```diff
@@ -85,22 +85,45 @@ function copy_files() {
...
+    BAZEL_CACHE_FLAGS="--remote_cache=${BAZEL_REMOTE_CACHE}"
+    if [[ -n ${BAZEL_EXPERIMENTAL_REMOTE_DOWNLOADER} ]]; then
+      BAZEL_CACHE_FLAGS+=" --experimental_remote_downloader=${BAZEL_EXPERIMENTAL_REMOTE_DOWNLOADER}"
+    fi
+  elif [[ -n ${BAZEL_DISK_CACHE} ]]; then
+    BAZEL_CACHE_FLAGS+="--disk_cache=${BAZEL_DISK_CACHE}"
+  fi
+
+  # Fetch emsdk and net_zlib dependencies
+  bazel --output_base="${OUTPUT_BASE}" fetch @emsdk//:BUILD @net_zlib//:README
```
Comment (jwendell) [inline maistra/scripts/update-deps.sh]: Do we still need this? I don't think so...

File: maistra/scripts/update-envoy-sha.sh [hunk 2]
```diff
@@ -29,12 +29,13 @@ function init() {
 
 function get_envoy_sha() {
   local branch
-  branch="${BRANCH:-$(git symbolic-ref --quiet --short HEAD)}"
+  branch="release/v1.28"  
+  # TODO: envoy-openssl and proxy should have the same branch
```
Comment (jwendell) [inline maistra/scripts/update-envoy-sha.sh]: They probably shouldn't. Envoy and Istio/Proxy are different projects, with different branches.

So, it's safe to remove this TODO comment. However, I'd keep the reuse of `BRANCH` variable if it's set:
```
branch="${BRANCH:-release/v1.28}"
```

File: .bazelrc [hunk 3]
```diff
@@ -49,6 +49,8 @@ build:debug -c dbg
 build --cxxopt -Wformat
 build --cxxopt -Wformat-security
 
+build --linkopt=-lstdc++fs
```
Comment (jwendell) [inline .bazelrc]: Do we really need this? If yes, what about moving it to `maistra/bazelrc`?

File: envoy.bazelrc [hunk 4]
```diff
@@ -181,12 +181,13 @@ build:clang-msan --test_env=MSAN_SYMBOLIZER_PATH
...
 
-# Clang with libc++
+# Clang with libstdc++
+build:libc++ --copt=-DGLIBCXX_USE_CXX11_ABI=0
 build:libc++ --config=clang
-build:libc++ --action_env=CXXFLAGS=-stdlib=libc++
-build:libc++ --action_env=LDFLAGS=-stdlib=libc++
-build:libc++ --action_env=BAZEL_CXXOPTS=-stdlib=libc++
-build:libc++ --action_env=BAZEL_LINKLIBS=-l%:libc++.a:-l%:libc++abi.a
+build:libc++ --action_env=CXXFLAGS=-stdlib=libstdc++
```
Comment (jwendell) [inline envoy.bazelrc:187]: I think we could achieve the same result using another config, not overriding/replacing this one.

File: maistra/scripts/update-envoy-sha.sh (hunk 2 above)
Comment (dcillera) [reply maistra/scripts/update-envoy-sha.sh]: I agree

File: .bazelrc (hunk 3 above)
Comment (dcillera) [reply .bazelrc]: Not sure it's really needed. It had been an attempt to clear some errors solved in other ways.

File: envoy.bazelrc (hunk 4 above)
Comment (dcillera) [reply envoy.bazelrc:187]: I guess something like: 
build: libcstd++ ...
and then replacing libc++ with that

Comment (jwendell) [reply envoy.bazelrc:187]: In the past we only had to make these changes in this file: https://github.com/maistra/proxy/commit/88766dfec7b870fab89678e2c313a056d1ca05f2#diff-4a029f39af2e2351c97f028719cdd15daa80be12172dcc67c7fdb0cf12f66ef2

Comment (dcillera) [reply envoy.bazelrc:187]: I tried the same but it didn't work. 
 libcstd++  is needed by the bSSL layer.

**Review by jwendell — APPROVED** [review]

**Pull request merged**